### DISCUSSION TITLE
Can I fetch data from protected schema?

### DISCUSSION THREAD

**Discussion by hooray0118** [supabase/supabase#8716] [Questions] [answered]
As I said above, I'd like to fetch data from storage schema. If possible, how can I do that?

**Comment by GaryAustin1** [comment, marked as answer]
You need to set schema in the init for the/a client instance if you are using a client library.
https://supabase.com/docs/reference/javascript/initializing#api-schemas
This works as long as the schema is listed in the dashboard API settings as being exposed (storage is by default).
There is also a header you can set if using the REST API directly.

**Comment by steve-chavez** [comment]
Alternatively you can create a SQL view in your public schema that refers to a storage schema table(only expose the columns that you need).

**Comment by J0** [comment]
Hey @hooray0118 ,

 Thanks for the query. Both @steve-chavez and @GaryAustin1 's answers should allow one to fetch data from the schema. Let us know if you face any further issues!
